app/collections/grade.js

```diff
@@ -1,77 +1,69 @@
-import {
-    saveFailed,
-    unableToRetrieve,
-    updateFailed,
-    deletionFailed
-} from '../notifications/general';
-
-const Datastore = require('nedb');
-const electron = require('electron');
-const path = require('path');
-
-const userDataPath = (electron.app || electron.remote.app).getPath('userData');
-const collectionsPath = path.join(userDataPath, 'collections');
-
-const Grade = new Datastore({
-    filename:              path.join(collectionsPath, 'grade.db'),
-    autoload:              true,
-    corruptAlertThreshold: 1,
-    timestampData:         true
-});
-
-export const updateGradeData = (data, id) => {
-    const {grade, examId, studentId, date, weight} = data;
-    Grade.update(
-        {_id: id},
-        {grade, examId, studentId, date, weight},
-        {},
-        err => {
-            if (err) {
-                updateFailed();
-            }
-        }
-    );
-};
-export const addGradeData = data => {
-    Grade.insert(data, (error, doc) => {
-        if (error) {
-            saveFailed();
-        }
-
-        return doc;
-    });
-};
-
-export const getAllGrades = () => new Promise(resolve => Grade.find({}, (err, docs) => {
-    if (err) {
-        unableToRetrieve();
+import connectionToDB from './connectionSetup';
+import {displayToast} from '../notifications';
+
+const Grade = connectionToDB('grade');
+
+export async function updateGradeData(data, id) {
+    try {
+        await Grade.update({_id: id}, {...data}, {});
```

Comment:
@hcuffy why are you de structuring data here?

Reply:
removed => fixed